src/rgen/Microsoft.Macios.Bindings.Analyzer/BindingTypeSemanticAnalyzer.cs

```diff
@@ -38,35 +35,17 @@ public override void Initialize (AnalysisContext context)
 	}
 
 	void AnalysisContext (SyntaxNodeAnalysisContext context)
-	{
-		// only care about classes
-		if (context.Node is not ClassDeclarationSyntax classDeclarationNode)
-			return;
+		=> this.AnalyzeBindingType (context);
 
-		var classSymbol = context.SemanticModel.GetDeclaredSymbol (classDeclarationNode);
-		if (classSymbol is null)
-			return;
+	public ImmutableArray<Diagnostic> Analyze (PlatformName _, ClassDeclarationSyntax declarationNode, INamedTypeSymbol symbol)
+	{
+		if (declarationNode.Modifiers.Any (x => x.IsKind (SyntaxKind.PartialKeyword)))
+			return [];
 
-		var boundAttributes = classSymbol.GetAttributes ();
-		if (boundAttributes.Length == 0) {
-			return;
-		}
+		var diagnostic = Diagnostic.Create (RBI0001,
```

Comment:
Add the actual error message as a comment:

```suggestion
		var diagnostic = Diagnostic.Create (RBI0001, // Binding types should be declared as partial classes.
```